agent/app/agent.go

```diff
@@ -145,6 +145,9 @@ func newAgent(
 		metadataManager = containermetadata.NewManager(dockerClient, cfg)
 	}
 
+	resource := resources.New()
+	resource.ApplyConfigDependencies(cfg)
```

Comment:
shouldn't this be happening only if cgroup feature is enabled?

Reply:
That logic is within resources, I believe.

The idea was that 'resources' could hold on to more things than just cgroups.

Reply:
i can put it behind after the `agent.cfg.TaskCPUMemLimit.Enabled` check.